### PR TITLE
Update CL metadata in dm-reader

### PR DESCRIPTION
**Pull request by jlaehne**
After fixing the metadata structure in LumiSpy, the mapping in the DM reader has to be updated. Implements the structure proposed in https://github.com/LumiSpy/lumispy/pull/109 (and should be merged only after tis PR).

Furthermore, an SEM as microscope is now properly detected and the metadata set accordingly.

Follow up on #2590, where this mapping was first introduced. AS #2590 is still unreleased, this redesign should be included in the v1.7 release.

### Progress of the PR
- [x] handle SEM as possible microscope instead of TEM,
- [x] update metadata according to https://github.com/LumiSpy/lumispy/pull/109,
- [x] add a changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] update tests,
- [x] ready for review.

### CODE REVIEW

File: hyperspy/io_plugins/digital_micrograph.py [hunk 1]
```diff
@@ -1001,59 +1025,82 @@ def get_mapping(self):
                     "Acquisition_instrument.TEM.Detector.EDS.real_time",
                     None),
             })
-        elif self.signal_type == "CL":
+        elif self.signal_type == "CL" or ("Acquisition" in image_tags_dict.keys() and "Monarc_Spectrometer" in image_tags_dict["Acquisition"].keys()):
```
Comment (ericpre) [inline hyperspy/io_plugins/digital_micrograph.py]: Include this in https://github.com/hyperspy/hyperspy/blob/d42c03d5986c649d2121fa34cfe361e9506e5a86/hyperspy/io_plugins/digital_micrograph.py#L627-L638?

File: hyperspy/io_plugins/digital_micrograph.py [hunk 2]
```diff
@@ -858,6 +858,25 @@ def _get_EELS_exposure_time(self, tags):
...
 
+    def _get_CL_detector_type(self, tags):
+        if "Acquisition_Mode" in tags and tags["Acquisition_Mode"] == "Parallel dispersive":
+            return "CCD"
+        elif "Acquisition_Mode" in tags and tags["Acquisition_Mode"] == "Serial dispersive":
+            return "PMT"
+        else:
+            _logger.info("CL detector type can't be read.")
+
+    def _get_CL_exposure_time(self, tags):
```
Comment (ericpre) [inline hyperspy/io_plugins/digital_micrograph.py]: Make it more generic and merge it with `_get_EELS_exposure_time`? https://github.com/hyperspy/hyperspy/blob/d42c03d5986c649d2121fa34cfe361e9506e5a86/hyperspy/io_plugins/digital_micrograph.py#L847-L859

**Review by ericpre — COMMENTED** [review]
Looks good, I have left some suggestions for improvement.

**Comment by jlaehne** [comment]
Thanks @ericpre. Two very good points that I took up.

Looking if there are easy ways to increase coverage, I realized that the functions `find_next_tag` and `find_next_data_tag` are not called anywhere and seem to be remnants from some early implementation. Any reason against removing them?

Also, having a file with complex data would do a big job for the coverage.

**Review by ericpre — APPROVED** [review]

**Comment by ericpre** [comment]
> Looking if there are easy ways to increase coverage, I realized that the functions `find_next_tag` and `find_next_data_tag` are not called anywhere and seem to be remnants from some early implementation. Any reason against removing them?

I guess, this have been used for debugging purposes but this is not documented, so I agree that this is fine to tidy it up.

**Pull request merged**